Cubical/Homotopy/Group/Pi4S3/BrunerieIso.agda

```diff
@@ -0,0 +1,526 @@
+                → (G→H : GroupHom G H) (G₂→H₂ : GroupHom G₂ H₂)
+                   (H→L : GroupHom H L) (H₂→L₂ : GroupHom H₂ L₂)
+                   (L→R : GroupHom L R)
+                → Exact4 G H L R G→H H→L L→R
+                → PathP (λ i → GroupHom (G≡G₂ i) (H≡H₂ i)) G→H G₂→H₂
+                → PathP (λ i → GroupHom (H≡H₂ i) (L≡L₂ i)) H→L H₂→L₂
+                → Exact4 G₂ H₂ L₂ UnitGr G₂→H₂ H₂→L₂ (→UnitHom L₂)
+transportExact4 {G = G} {G₂ = G₂} {H = H} {H₂ = H₂} {L = L} {L₂ = L₂} {R = R} =
+  J4 (λ G₂ H₂ L₂ R G≡G₂ H≡H₂ L≡L₂ Unit≡R
+                → (G→H : GroupHom G H) (G₂→H₂ : GroupHom G₂ H₂)
+                   (H→L : GroupHom H L) (H₂→L₂ : GroupHom H₂ L₂)
+                   (L→R : GroupHom L R)
+                → Exact4 G H L R G→H H→L L→R
+                → PathP (λ i → GroupHom (G≡G₂ i) (H≡H₂ i)) G→H G₂→H₂
+                → PathP (λ i → GroupHom (H≡H₂ i) (L≡L₂ i)) H→L H₂→L₂
+                → Exact4 G₂ H₂ L₂ UnitGr G₂→H₂ H₂→L₂ (→UnitHom L₂))
+      (λ G→H G₂→H₂ H→L H₂→L₂ L→R ex pp1 pp2
+        → J4 (λ G₂→H₂ H₂→L₂ (x : Unit) (y : Unit)
+                 pp1 pp2 (_ : tt ≡ x) (_ : tt ≡ x)
+             → Exact4 G H L UnitGr G₂→H₂ H₂→L₂ (→UnitHom L))
+               ex G₂→H₂ H₂→L₂ tt tt pp1 pp2 refl refl )
+      G₂ H₂ L₂ R
+  where
+  J4 : ∀ {ℓ ℓ₂ ℓ₃ ℓ₄ ℓ'} {A : Type ℓ}
+       {A₂ : Type ℓ₂} {A₃ : Type ℓ₃} {A₄ : Type ℓ₄}
+       {x : A} {x₂ : A₂} {x₃ : A₃} {x₄ : A₄}
+       (B : (y : A) (z : A₂) (w : A₃) (u : A₄)
+       → x ≡ y → x₂ ≡ z → x₃ ≡ w → x₄ ≡ u → Type ℓ')
+    → B x x₂ x₃ x₄ refl refl refl refl
+    → (y : A) (z : A₂) (w : A₃) (u : A₄)
+       (p : x ≡ y) (q : x₂ ≡ z) (r : x₃ ≡ w) (s : x₄ ≡ u)
+    → B y z w u p q r s
+  J4 {x = x} {x₂ = x₂} {x₃ = x₃} {x₄ = x₄} B b y z w u =
+    J (λ y p → (q : x₂ ≡ z) (r : x₃ ≡ w) (s : x₄ ≡ u) →
+      B y z w u p q r s)
+      (J (λ z q → (r : x₃ ≡ w) (s : x₄ ≡ u) → B x z w u refl q r s)
+        (J (λ w r → (s : x₄ ≡ u) → B x x₂ w u refl refl r s)
+          (J (λ u s → B x x₂ x₃ u refl refl refl s) b)))
+
+
+{-
+We will now instantiate Blakers Massey to get a square:
+
+         fold∘W
+ S³ --------------> S²
+ |\              ↗ |
+ |  \         ↗    |
+ |    \    ↗       |
+ |      X           |  inr
+ |    /             |
+ |   /              |
+ |  /               |
+ v /                v
+ 1 -----------> coFib fold∘W
+         inl
+
+where X is the pullback of inl and inr and the map S³ → X is
+surjective on π₃. This will give us a sequence
+π₃ S³ --ᶠ→ π₃ S² → π₃ (coFib fold∘W) → π₂ X ≅ 0, where f is
+the function induced by fold∘W. From this, we can deduce that
+π₃ (coFib fold∘W) ≅ ℤ/ f(1) where f(1) is interpreted as an
+integer via the isos π₃ S³ ≅ π₃ S² ≅ ℤ
+
+(Recall that π₃(coFib fold∘W) ≅ π₄S³)
+
+For clarity:
+X, above will have two names (via a trivial iso) below:
+
+TotalPushoutPath× (the version the falls out of BM)
+P = fiber inr (Same name as in Brunerie's prop 4.3.4.)
+-}
+
+
+-- Before instantiating, we need to show that
+-- any map S³ → S² is 0-connected
```

Comment:
This should probably be moved somewhere else, but I'm not sure so unless you can think of a good place let's leave it

Reply:
I'd vote for leaving it for now. If we want to generalise it in the future, it would probably belong in `Connected`

Reply:
Ok, sounds good